### PR TITLE
add page on attestations to pos section

### PR DESCRIPTION
**Pull request by jmcook1186**
## Description

Adds page on attestations to `developers/docs/consensus-mechanisms/pos`.
Uses kb.beaconchain attestation page as base but pretty heavily modified, including new diagram.

## Related Issue

#2277

### CODE REVIEW

**Review by wackerow — COMMENTED** [review]
❤️ love to see this =) 

Committed a couple small adjustments during my first quick pass... 

Our nice new image =) Love this, but two concerns, 1) translations and 2) dark mode. 

image

The later is a blocker, but I'm sure can easily be fixed... is this in Figma? @nloureiro was this you?

Will try to consider options for intl support here in the meantime

**Comment by jmcook1186** [comment]
damn, dark mode ofc. I whipped that image up bc I thought the one in the original page was a bit scrappy for e.org. @nloureiro can you help make this e.org compatible?

**Comment by nloureiro** [comment]
> damn, dark mode ofc. I whipped that image up bc I thought the one in the original page was a bit scrappy for e.org. @nloureiro can you help make this e.org compatible?

Yes, I can try.

What is the page? Is it this one? https://ethereumorgwebsitedev01-jcbeaconchainattestations.gtsb.io/en/developers/docs/consensus-mechanisms/pos

For translation, @wackerow any idea? should we make this diagram a component with text in html and the rest svg?

**Comment by wackerow** [comment]
@nloureiro https://ethereumorgwebsitedev01-jcbeaconchainattestations.gtsb.io/en/developers/docs/consensus-mechanisms/pos/attestations

As for translations, this is a similar problem that plagues our developer docs... I think we should consider this as a separate initiative (possibly a future epic cc: @samajammin) to fix all of these images that use embedded English. In the meantime, if we just keep the Figma you use handy, we can always re-export without text in the future.

File: src/content/developers/docs/consensus-mechanisms/pos/attestations/index.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+- `index`: A number that identifies which committee the validator belongs to in a given slot
+- `beacon_block_root`: Root hash of the block the validator sees at the head of the chain (the result of applying the fork-choice algorithm)
+- `source`: Part of the finality vote indicating what the validators see as the most recent justified block
+- `target`: Part of the finality vote indicating what the validators see as the first block in the current epoch
+
+Once the `data` is built, the validator can flip the bit in `aggregation_bits` corresponding to their own validator index from 0 to 1 to show that they participated.
+
+Finally, the validator signs the attestation and broadcasts it to the network.
+
+### Aggregated Attestation {#aggregated-attestation}
```
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/pos/attestations/index.md:32]:
```suggestion
### Aggregated attestation {#aggregated-attestation}
```

File: src/content/developers/docs/consensus-mechanisms/pos/attestations/index.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+In each epoch a validator in each subnet is selected to be the `aggregator`. The aggregator collects all the attestations it hears about over the gossip network that have equivalent `data` to its own. The sender of each matching attestation is recorded in the `aggregation_bits`. The aggregator then broadcasts the attestation aggregate to the wider network.
+
+When a validator is selected to be a block proposer they package aggregate attestations from the subnets up to the latest slot in the new block.
+
+### Attestation Inclusion Lifecycle {#attestation-inclusion-lifecycle}
```
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/pos/attestations/index.md:40]: Please use sentence case for titles :-)

```suggestion
### Attestation inclusion lifecycle {#attestation-inclusion-lifecycle}
```

File: src/content/developers/docs/consensus-mechanisms/pos/attestations/index.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
+---
+title: Attestations
+description: A description of attestations on PoS Ethereum.
+lang: en
+sidebar: true
+---
+
+A validator is expected to create, sign and broadcast an attestation during every epoch. This page outlines what these attestations look like and how they are processed and communciated between consensus clients.
+
+## What is an attestation? {#what-is-an-attestation}
```
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/pos/attestations/index.md:10]: Can we add pre-requisites here

File: src/content/developers/docs/consensus-mechanisms/pos/attestations/index.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
+---
+title: Attestations
+description: A description of attestations on PoS Ethereum.
```
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/pos/attestations/index.md:3]:
```suggestion
description: A description of attestations on proof-of-stake Ethereum.
```

**Review by minimalsm — CHANGES_REQUESTED** [review]
This has been hanging around for a while.

To get this merged quickly, could I suggest:

- We add a non-transparent background to the image for now
- We create a card to revisit the image (probably as an epic?)
- We add a pre-requisites section

**Review by minimalsm — APPROVED** [review]

**Pull request merged**